src/Storages/StorageTableFunction.h

```diff
@@ -66,9 +66,7 @@ class StorageTableFunctionProxy final : public StorageProxy
     String getName() const override
     {
         std::lock_guard lock{nested_mutex};
-        if (nested)
-            return nested->getName();
-        return StorageProxy::getName();
+        return getNested()->getName();
```

Comment:
> What is the reasoning behind it?

Without these changes, the first selection  of the `engine` row from system.tables will return `StorageProxy`, instead of the actual engine name, because nested class does not created. And also does not matter remote source accessible or not.
Example:
```
CREATE table test_pg (id Int, value Int) AS postgresql('localhost:5432', 'test_db', 'test_table', 'test1', '123456')
SELECT engine FROM system.tables WHERE name='test_pg'\G

Row 1:
──────
engine: StorageProxy
```

Reply:
>Second, if it stays as is, you don't need to lock the nested_mutex. 

Yeh, will fix this

Reply:
AFAIR it was intentionally done like this to avoid constructing a nested storage just to get its type. mostly because we may fail to construct it because the remote source may be unavailable

Reply:
Reverted it in the last commit